packages/auth-manager/src/connection/models/connectionManager.ts

```diff
@@ -89,4 +95,27 @@ export class ConnectionManager {
       return connectionRepo.save({ ...connection, version: maxVersion + 1 });
     });
   }
+
+  private async handleToken(connection: IConnection, transactionKeyRepo: KeyRepository): Promise<string> {
+    if (connection.token !== '') {
+      return connection.token;
+    }
+
+    const key = (await transactionKeyRepo.getLatestKeys()).find((key) => key.environment === connection.environment);
+
+    if (key === undefined || key.privateKey === undefined) {
+      this.logger.warn({
+        msg: 'no private key found for connection, could not create token',
+        connection: { name: connection.name, environment: connection.environment },
+      });
+      return '';
+    }
+
+    try {
+      return await generateToken(key.privateKey as JWK, connection.name, key.privateKey.kid);
+    } catch (error) {
+      this.logger.error({ msg: 'could not generate token', error });
+      return '';
+    }
```

Comment:
I don't like the best-effort approach, better having a flag specifying that token is not a must.

Reply:
Added a query param to specify the requested behaviour.